t/23_utils_git.t

```diff
@@ -0,0 +1,38 @@
+use strict;
+use warnings;
+use testapi;
+use Test::MockModule;
+use Test::Exception;
+use Test::More;
+
+use Utils::Git;
+
+subtest '[git_clone] Compose command' => sub {
+    my $mock_git = Test::MockModule->new('Utils::Git', no_auto => 1);
+    my @calls;
+    $mock_git->redefine(assert_script_run => sub { @calls = $_[0]; return 0; });
+    $mock_git->redefine(record_info => sub { return 0; });
+    $mock_git->redefine(upload_logs => sub { return 0; });
+    my $repository = 'https://github.com/os-autoinst/os-autoinst-distri-opensuse';
```

Comment:
```suggestion
    my $repository = 'FISH_N_CHIPS';
```

Be creative ;-) 

The point is that tested function does not care (for the moment) about the url format, its just a string so give it just a string

Reply:
What about `https://****hub.com/spicy_code/global_variables_only` ? :D